### PR TITLE
📅 End meeting for all

### PR DESCRIPTION
**Pull request by PVince81**
- [x] backend impl (based on read-only mode kicking out users)
- [x] PHP tests
- [x] frontend button
- [x] vuex store action + backend call
- [x] only moderators or guest moderators can end for all
- [x] also send signaling message to kick people out more quickly ?
- [x] system message for "X has ended the call" instead of "you left the call"
- [x] Jest tests
- [x] Test with Android
- [x] Test with iOS

---

- [x] @marcoambrosini for the button design
- [x] @danxuliu for terminating the call screen as another user in `src/utils/webrtc/webrtc.js` (ref https://github.com/nextcloud/spreed/pull/5981#issuecomment-907206782 )
- [x] @nickvergessen for changing the system message and call summary

---

- [x] TEST: without HPB: end meeting for all :no_entry_sign:  does not kick out user from another browser
- [x] TEST: HPB: end meeting for all :no_entry_sign:  does not kick out user from another browser

Fixes https://github.com/nextcloud/spreed/issues/5395

### CODE REVIEW

**Comment by PVince81** [comment]
note: the current button is a hack of an Actions menu:
image

we might need to create a new component for that later, or extend the Actions component to allow for a label and also changing the default colors ?

**Comment by nickvergessen** [comment]
From my pov we should go for a dedicated meeting setting section in the right sidebar first.
So there is quick access to mute all, end call, end lobby, ...

File: lib/Controller/CallController.php [hunk 1]
```diff
@@ -152,15 +152,25 @@ public function updateCallFlags(int $flags): DataResponse {
...
 	 */
-	public function leaveCall(): DataResponse {
+	public function leaveCall(bool $all): DataResponse {
 		$session = $this->participant->getSession();
 		if (!$session instanceof Session) {
 			return new DataResponse([], Http::STATUS_NOT_FOUND);
 		}
 
-		$this->participantService->changeInCall($this->room, $this->participant, Participant::FLAG_DISCONNECTED);
+		if ($all) {
```
Comment (nickvergessen) [inline lib/Controller/CallController.php:164]: Check for moderator permissions 🙈

**Comment by PVince81** [comment]
> From my pov we should go for a dedicated meeting setting section in the right sidebar first.
> So there is quick access to mute all, end call, end lobby, ...

Ah yes, I remember now that you said it before. Got distracted by @marcoambrosini's latest proposal.

**Comment by nickvergessen** [comment]
Assigning:
* @marcoambrosini for the button design
* @danxuliu for terminating the call screen as another user in `src/utils/webrtc/webrtc.js` (ref https://github.com/nextcloud/spreed/pull/5981#issuecomment-907206782 )
* @nickvergessen for changing the system message and call summary

**Comment by nickvergessen** [comment]
activity is fixed

**Comment by marcoambrosini** [comment]
So I discussed this with @jancborchardt and @nimisha-vijay and for now we're going to go with the button design that's currently implemented.

**Comment by nickvergessen** [comment]
> Test with Android

Stays on the call screen: https://github.com/nextcloud/talk-android/issues/1660

> Test with iOS

Stays on the call screen: https://github.com/nextcloud/talk-ios/issues/658

**Comment by nickvergessen** [comment]
Time to get it in then?

Rebased to fix up

**Comment by Ivansss** [comment]
Could we add a capability here for the clients?

**Review by danxuliu — COMMENTED** [review]
Is the last commit right? `Sounds.playLeave(true)` is still called both when the user leaves the call locally and when the meeting is ended remotely.

**Comment by nickvergessen** [comment]
didn't ment to have that pushed. sorry, will remove again

**Comment by nickvergessen** [comment]
> Could we add a capability here for the clients?

Available with `publishing-permissions` capability and only works for moderators

**Pull request merged**